saleor/order/models.py

```diff
@@ -130,9 +130,17 @@ class Order(ModelWithMetadata):
         related_name="orders",
         on_delete=models.SET_NULL,
     )
+    collection_point = models.ForeignKey(
+        "warehouse.Warehouse",
+        blank=True,
+        null=True,
+        related_name="orders",
+        on_delete=models.SET_NULL,
```

Comment:
I don't recall if we discussed it - do we somehow handle the case when a warehouse is deleted from the system, but it was selected as a collection point in some existing orders?

Reply:
No, we haven't discuss it, this could be a tough one.  Do you think that,  for instance `on_delete=models.PROTECT` will be better?

Reply:
Based on our short conversation, I have add an `collection_point_name` character field, similar to `shipping_method_name`